llpc/lower/llpcSpirvLowerMath.h

```diff
@@ -78,6 +78,17 @@ class SpirvLowerMathConstFolding : public SpirvLowerMath, public llvm::PassInfoM
   llvm::Function *getEntryPoint();
 };
 
+// =====================================================================================================================
+// SPIR-V lowering operations to adjust fast math flags.
+class SpirvLowerMathPrecision : public SpirvLowerMath, public llvm::PassInfoMixin<SpirvLowerMathPrecision> {
```

Comment:
Let's please not have passes inherit from each other. What's the motivation? Access to `disableFastMath`? It would be better to extract that function then.

Reply:
`SpirvLowerMath` is a super-class to passes `SpirvLowerMathConstFolding` and `SpirvMathFloatOp` already, so I simply reused the existing pattern.  I assume this was done to reuse implementation details.
Are you asking me to split all of these down and remove `SpirvLowerMath`?

Reply:
To further clarify, `SpirvLowerMath` is not a pass, it is a helper class used by the _Math_ passes, so there is no inheritance between passes.
Not much of `SpirvLowerMath` is used by this new pass, so I can separate it if that is the preference.

Reply:
Yes, please do that. Using inheritance like that is a pattern that we should move away from. The situation has recently improved somewhat in the raytracing passes, for example.

Reply:
Done, assuming you were not asking me to remove all inheritance from `SpirvLower`.